tests/test_common.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file test_common.h
+ * @author Irfan
+ * @brief common header file for all tests to facilitate uniform logging format
+ *
+ * @copyright
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef SR_TEST_COMMON_H
+#define SR_TEST_COMMON_H
+
+#include <pthread.h>
+#include <assert.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include "sysrepo.h"
+#include "tests/config.h"
+
+/**
+ * Only function that needs to be called from test code
+ * Initializes callback and facilitates logging to stderr
+ * with timestamps and thread-id
+ */
+static void test_log_init();
+
+#define LOG_INF(...) _test_log(SR_LL_INF, __VA_ARGS__)
+#define LOG_ERR(...) _test_log(SR_LL_ERR, __VA_ARGS__)
+#define LOG_WRN(...) _test_log(SR_LL_WRN, __VA_ARGS__)
```

Comment:
Am I missing something or these macros are not used at all?

Reply:
I am planning on replacing the various printf, fprintf in the test code in future commits. 
For instance please see below code. 

```c
#define sr_assert(cond) if (!(cond)) { fprintf(stderr, "\"%s\" not true\n", #cond); sr_assert_line(); abort(); }

#define sr_assert_line() printf("[   LINE    ] --- %s:%d: error: Failure!\n", __FILE__, __LINE__)

#define sr_assert_true(cond) if (!(cond)) { fprintf(stderr, "\"%s\" not true\n", #cond); sr_assert_line(); return 1; }

#define sr_assert_int_equal(val1, val2) { \
    int ret1, ret2; \
    ret1 = val1; ret2 = val2; \
    if (ret1 != ret2) { fprintf(stderr, "%d != %d\n", ret1, ret2); sr_assert_line(); return 1; } }
```

Reply:
And also add logs to the test itself, explaining what's being done in case of complex tests, can be done using LOG_INF("Starting some new test explanation");

Reply:
These assert are specific for the multi-process test, you have no option to change logging for the other `cmocka` tests. Also, not sure some INF messages are a good idea, there is already lots and lots of output.

Reply:
I understand your concern. I will change the commit to be limited to a common logging callback for all the tests. Is that okay?

Reply:
Yes, I have no problem with that.